### ISSUE TITLE
Frontend - Non-existing pages are answered with a HTTP 200 OK

### ISSUE THREAD

**Issue by porst17** [cloudradar-monitoring/rport#471]
If have set up RPort 0.6.0 using the installer script from https://get.rport.io
Now I wanted to secure the server using fail2ban following https://oss.rport.io/docs/no10-securing-the-server.html#using-fail2ban-for-additional-security

HTTP request to non-existing pages are properly answered with a `404 NOT FOUND`:
```
$ curl -I http://<RPORT_SERVER>/phpMyadmin/index.php
HTTP/1.1 404 Not Found
...
```

However, doing the same with HTTPS results in a "This page could not be found" HTML page with status code `200 OK` ("Soft 404"):
```
$ curl -I https://<RPORT_SERVER>/phpMyadmin/index.php
HTTP/2 200
...
```

I know that most scanner target HTTP because HTTPS has too much overhead, but is this intended behavior? Other sites, e.g. GitHub display a proper 404 error page (with `404 NOT FOUND`) in this case and if I am not mistaken, soft 404 is discouraged in general.

**Comment by thorstenkramm** [comment]
@porst17 
Yes. This behavior is correct and required by Vue.js. 
Single page apps like ours require a rewrite of all paths to index.html.
I would not consider it as a security drawback. If you access `https://<RPORT_SERVER>/phpMyadmin/index.php` you will get back the index.html. Any attempt to send a POST request to abuse potential security holes will run into the void.

Returning 404 is doable because the URLs of a vue app follow a known pattern. But I'm not confident how it would increase the security.

**Comment by porst17** [comment]
Well, the [RPort documentation proposes securing the RPort server by adding fail2ban to block attackers that scan for vulnerable sites](https://oss.rport.io/docs/no10-securing-the-server.html#using-fail2ban-for-additional-security). Your proposed fail2ban filter looks for 404 error codes in the RPort log file. This works for the HTTP server, because it properly returns 404 for non-existing sites. However, the server that's responsible for the API and the frontend does not reply with 404 in such cases which prevents the above approach of banning scanners through fail2ban.

I just checked my logs and I have about the same number of scans on the RPort server (configured to run on port 80 in the `[server]` section) as on the API/frontend server (configured to run on port 443 in `[api]` section). That's why I think it would make sense to enable `fail2ban` for the API/frontend server as well.

**Comment by thorstenkramm** [comment]
@porst17 Fair enough. It makes sense to return 404 whenever possible to identify scanners with the goal to ban them with fail2ban. I'll file a task in our backlog.

**Comment by porst17** [comment]
Looks like this has been implemented in [v0.7.0](https://github.com/cloudradar-monitoring/rport/releases/tag/0.7.0). Thanks a ton! 🎉 

I already updated my server and it seems to works as expected. (I accidentally banned myself during testing. 🤣 )
